app/assets/javascripts/admin/addon/templates/admin-badges/show.hbs

```diff
@@ -280,10 +285,31 @@
         @showTitle={{false}}
         @name="show_posts"
         @disabled={{this.readOnly}}
+        @onSet={{this.onSetShowPosts}}
         as |field|
       >
         <field.Checkbox />
       </group.Field>
+
+      <group.Field
+        @title={{i18n "admin.badges.post_header"}}
+        @showTitle={{false}}
+        @name="post_header"
+        @disabled={{this.disableBadgeOnPosts}}
```

Comment:
It seems like this is only run once, on initial render. It isn't being re-run when the other checkboxes are changed.

Reply:
Ah yes it's on my list to investigate, will look thanks for reminder :+1:

Reply:
Fixed in: https://github.com/discourse/discourse/pull/29738

Reply:
That fixed it here, thanks @jjaffeux!